Removes support for Java 14 features and thus allows build with Java 11.

Reverts parts org.eclipse.jdt.compiler.apt/src/org/eclipse/jdt/* parts from the original commit:

https://git.eclipse.org/c/jdt/eclipse.jdt.core.git/commit/?id=89f366a91c008f7ff476d2cde4a14dc21ddaf6fa

--- org/eclipse/jdt/internal/compiler/apt/model/Factory.java
+++ org/eclipse/jdt/internal/compiler/apt/model/Factory.java
@@ -328,7 +328,6 @@
 			case ANNOTATION_TYPE :
 			case INTERFACE :
 			case CLASS :
-			case RECORD :
 				// for type
 				decodeModifiers(result, modifiers, new int[] {
 					ClassFileConstants.AccPublic,
--- org/eclipse/jdt/internal/compiler/apt/model/TypeElementImpl.java
+++ org/eclipse/jdt/internal/compiler/apt/model/TypeElementImpl.java
@@ -33,7 +33,6 @@
 import javax.lang.model.element.Name;
 import javax.lang.model.element.NestingKind;
 import javax.lang.model.element.PackageElement;
-import javax.lang.model.element.RecordComponentElement;
 import javax.lang.model.element.TypeElement;
 import javax.lang.model.element.TypeParameterElement;
 import javax.lang.model.element.VariableElement;
@@ -90,7 +89,6 @@
 				case INTERFACE :
 				case CLASS :
 				case ENUM :
-				case RECORD :
 					TypeElementImpl typeElementImpl = (TypeElementImpl) e;
 					Binding typeBinding = typeElementImpl._binding;
 					if (typeBinding instanceof SourceTypeBinding) {
@@ -110,7 +108,6 @@
 					break;
 				case ENUM_CONSTANT :
 				case FIELD :
-				case RECORD_COMPONENT :
 					VariableElementImpl variableElementImpl = (VariableElementImpl) e;
 					binding = variableElementImpl._binding;
 					if (binding instanceof FieldBinding) {
@@ -168,13 +165,6 @@
 				 enclosed.add(variable);
 			}
 		}
-		if (binding.isRecord()) {
-			RecordComponentBinding[] components = binding.components();
-			for (RecordComponentBinding comp : components) {
-				RecordComponentElement rec = new RecordComponentElementImpl(_env, comp);
-				enclosed.add(rec);
-			}
-		}
 		for (ReferenceBinding memberType : binding.memberTypes()) {
 			TypeElement type = new TypeElementImpl(_env, memberType, null);
 			enclosed.add(type);
@@ -183,23 +173,6 @@
 		return Collections.unmodifiableList(enclosed);
 	}
 
-	@Override
-    public List<? extends RecordComponentElement> getRecordComponents() {
-		if (_binding instanceof SourceTypeBinding) {
-			SourceTypeBinding binding = (SourceTypeBinding) _binding;
-			List<RecordComponentElement> enclosed = new ArrayList<>();
-			for (RecordComponentBinding comp : binding.components()) {
-				RecordComponentElement variable = new RecordComponentElementImpl(_env, comp);
-				enclosed.add(variable);
-			}
-			Collections.sort(enclosed, new SourceLocationComparator());
-			return Collections.unmodifiableList(enclosed);
-		}
-		// TODO: Add code for BinaryTypeBinding, which, as of now doesn't seem to contain components
-		return Collections.emptyList();
-    }
-
-	@Override
 	public List<? extends TypeMirror> getPermittedSubclasses() {
 		ReferenceBinding binding = (ReferenceBinding)_binding;
 		if (binding.isSealed()) {
@@ -264,9 +237,6 @@
 		if (refBinding.isEnum()) {
 			return ElementKind.ENUM;
 		}
-		else if (refBinding.isRecord()) {
-			return ElementKind.RECORD;
-		}
 		else if (refBinding.isAnnotationType()) {
 			return ElementKind.ANNOTATION_TYPE;
 		}
--- org/eclipse/jdt/internal/compiler/apt/model/ElementsImpl.java
+++ org/eclipse/jdt/internal/compiler/apt/model/ElementsImpl.java
@@ -347,7 +347,6 @@
 			case CLASS :
 			case ENUM :
 			case INTERFACE :
-			case RECORD :
 				TypeElementImpl typeElementImpl = (TypeElementImpl) e;
 				ReferenceBinding referenceBinding = (ReferenceBinding)typeElementImpl._binding;
 				if (referenceBinding instanceof SourceTypeBinding) {
@@ -378,7 +377,6 @@
 					referenceContext = sourceMethod;
 				}
 				break;
-			case RECORD_COMPONENT :
 			case ENUM_CONSTANT :
 			case FIELD :
 				VariableElementImpl variableElementImpl = (VariableElementImpl) e;
@@ -584,7 +582,6 @@
 			case CLASS :
 			case ENUM :
 			case INTERFACE :
-			case RECORD :
 				TypeElementImpl typeElementImpl = (TypeElementImpl) type;
 				ReferenceBinding referenceBinding = (ReferenceBinding)typeElementImpl._binding;
 				return (PackageElement) _env.getFactory().newElement(referenceBinding.fPackage);
@@ -597,7 +594,6 @@
 				return (PackageElement) _env.getFactory().newElement(methodBinding.declaringClass.fPackage);
 			case ENUM_CONSTANT :
 			case FIELD :
-			case RECORD_COMPONENT :
 				VariableElementImpl variableElementImpl = (VariableElementImpl) type;
 				FieldBinding fieldBinding = (FieldBinding) variableElementImpl._binding;
 				return (PackageElement) _env.getFactory().newElement(fieldBinding.declaringClass.fPackage);
--- org/eclipse/jdt/internal/compiler/apt/model/TypesImpl.java
+++ org/eclipse/jdt/internal/compiler/apt/model/TypesImpl.java
@@ -138,7 +138,6 @@
 			break;
 		case FIELD:
 		case ENUM_CONSTANT:
-		case RECORD_COMPONENT:
 			typeMirror = findMemberInHierarchy(referenceBinding, elementImpl._binding, new MemberInTypeFinder() {
 				@Override
 				public TypeMirror find(ReferenceBinding typeBinding, Binding memberBinding) {
@@ -160,7 +159,6 @@
 		case ANNOTATION_TYPE:
 		case INTERFACE:
 		case CLASS:
-		case RECORD:
 			typeMirror = findMemberInHierarchy(referenceBinding, elementImpl._binding, new MemberInTypeFinder() {
 				@Override
 				public TypeMirror find(ReferenceBinding typeBinding, Binding memberBinding) {
--- org/eclipse/jdt/internal/compiler/apt/model/RecordComponentElementImpl.java
+++ /dev/null
@@ -1,54 +0,0 @@
-/*******************************************************************************
- * Copyright (c) 2020, 2021 IBM Corporation.
- *
- * This program and the accompanying materials
- * are made available under the terms of the Eclipse Public License 2.0
- * which accompanies this distribution, and is available at
- * https://www.eclipse.org/legal/epl-2.0/
- *
- * SPDX-License-Identifier: EPL-2.0
- *
- * Contributors:
- *     IBM Corporation - initial API and implementation
- *******************************************************************************/
-
-package org.eclipse.jdt.internal.compiler.apt.model;
-
-import javax.lang.model.element.ElementKind;
-import javax.lang.model.element.ElementVisitor;
-import javax.lang.model.element.ExecutableElement;
-import javax.lang.model.element.RecordComponentElement;
-
-import org.eclipse.jdt.internal.compiler.apt.dispatch.BaseProcessingEnvImpl;
-import org.eclipse.jdt.internal.compiler.lookup.MethodBinding;
-import org.eclipse.jdt.internal.compiler.lookup.RecordComponentBinding;
-import org.eclipse.jdt.internal.compiler.lookup.ReferenceBinding;
-import org.eclipse.jdt.internal.compiler.lookup.SourceTypeBinding;
-
-public class RecordComponentElementImpl extends VariableElementImpl implements RecordComponentElement {
-
-	protected RecordComponentElementImpl(BaseProcessingEnvImpl env, RecordComponentBinding binding) {
-		super(env, binding);
-	}
-
-	@Override
-	public ElementKind getKind() {
-		return ElementKind.RECORD_COMPONENT;
-	}
-
-	@Override
-	public ExecutableElement getAccessor() {
-		RecordComponentBinding comp = (RecordComponentBinding) this._binding;
-		ReferenceBinding binding = comp.declaringRecord;
-		if (binding instanceof SourceTypeBinding) {
-			MethodBinding accessor = ((SourceTypeBinding) binding).getRecordComponentAccessor(comp.name);
-			return new ExecutableElementImpl(_env, accessor);
-		}
-		return null;
-	}
-	@SuppressWarnings("preview")
-	@Override
-	public <R, P> R accept(ElementVisitor<R, P> visitor, P param) {
-		return visitor.visitRecordComponent(this, param);
-	}
-}
